staging/src/k8s.io/legacy-cloud-providers/vsphere/nodemanager.go

```diff
@@ -288,8 +288,6 @@ func (nm *NodeManager) GetNodeInfo(nodeName k8stypes.NodeName) (NodeInfo, error)
 //
 // This method is a getter but it can cause side-effect of updating NodeInfo objects.
 func (nm *NodeManager) GetNodeDetails() ([]NodeDetails, error) {
-	nm.registeredNodesLock.Lock()
-	defer nm.registeredNodesLock.Unlock()
 	var nodeDetails []NodeDetails
 
 	for nodeName, nodeObj := range nm.registeredNodes {
```

Comment:
ranging over registeredNodes in an unprotected way is not safe in combination with calls to addNode or removeNode

Reply:
You are right, completely missed this part, thanks! :100:  Now all operations will be performed on a copy of the `nm.registeredNodes` map.

Reply:
What are the performance implications of copying this map? What are the correctness implications of iterating over potentially stale data outside the lock?

Reply:
This lock is only set once there is an update requested for the `Node(s)`. If anything else needs to refresh the node info, it would do it the same way. Otherwise, under normal circumstances, the `nodeInfoMap` or `registeredNodes` map would be populated only once - it only accounts for changes outside of the `kubernetes` cluster management. So it is not critical to iterating over potentially stale data, but I would refer to `vSphere` contributors to confirm.

Reply:
The performance implications are minimal - it would actually speed up the process, if something else is waiting on the lock to do the full cycle - now it won't. It is just a `map[string]*v1.Node` copy, after all, executed mostly once there is a `kcm` pod restart or other resource change.